### PR TITLE
Add configmap json patch transformer plugin

### PR DESCRIPTION
**Pull request by dearchap**
This plugin allows configmap keys which have json content to be patched using json6902 spec. The plugin has used the patchJson6902 plugin as a baseline to incorporate all the relevant features.

### CODE REVIEW

**Comment by dearchap** [comment]
Just signed the CLA

**Comment by dearchap** [comment]
@mortent Any comments on this ?

File: plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	)
+	obj, err := m.GetById(id)
+	if err != nil {
+		return err
+	}
+	if obj.GetKind() != "ConfigMap" {
+		return fmt.Errorf("This patch cannot be applied to non configmap resources")
+	}
+
+	configMapData, gerr := obj.GetFieldValue("data")
```
Comment (Shell32-Natsu) [inline plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go:79]: Can you use `kyaml/yaml` library to manipulate resources? You can view the example in `api/builtins/PatchJson6902Transformer.go`

File: plugin/someteam.example.com/v1/configmappatchjsontransformer/go.mod [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+require (
+	github.com/evanphx/json-patch v4.5.0+incompatible
+	github.com/pkg/errors v0.8.1
+	sigs.k8s.io/kustomize/api v0.5.1
+	sigs.k8s.io/kustomize/kyaml v0.7.1
+	sigs.k8s.io/yaml v1.2.0
+)
+
+replace sigs.k8s.io/kustomize/api v0.5.1 => ../../../../api
```
Comment (Shell32-Natsu) [inline plugin/someteam.example.com/v1/configmappatchjsontransformer/go.mod:13]: api and kyaml versions need to be updated.

**Review by Shell32-Natsu — CHANGES_REQUESTED** [review]

File: plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go (hunk 1 above)
Comment (dearchap) [reply plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go:79]: @Shell32-Natsu I'm using the exact same imports as the builtin patchJson6902Transformer. That isnt sufficient ?

File: plugin/someteam.example.com/v1/configmappatchjsontransformer/go.mod (hunk 2 above)
Comment (dearchap) [reply plugin/someteam.example.com/v1/configmappatchjsontransformer/go.mod]: Updated.

File: plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go (hunk 1 above)
Comment (dearchap) [reply plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go:79]: @mortent

Comment (dearchap) [reply plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go:80]: @Shell32-Natsu Just updated to use kyaml

**Comment by dearchap** [comment]
@Shell32-Natsu Can you review ?

**Pull request closed without merging**